pkg/querier/frontend/frontend.go

```diff
@@ -50,20 +50,22 @@ var (
 type Config struct {
 	MaxOutstandingPerTenant int
 	MaxRetries              int
+	SplitQueriesByDay       bool
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxOutstandingPerTenant, "querier.max-outstanding-requests-per-tenant", 100, "Maximum number of outstanding requests per tenant per frontend; requests beyond this error with HTTP 429.")
-	f.IntVar(&cfg.MaxRetries, "querier.max-retries-per-request", 5, "Maximum number of retries for a single request; beyon this, the downstream error is returned.")
+	f.IntVar(&cfg.MaxRetries, "querier.max-retries-per-request", 5, "Maximum number of retries for a single request; beyond this, the downstream error is returned.")
+	f.BoolVar(&cfg.SplitQueriesByDay, "querier.split-queries-by-day", false, "Split queries by day and execute in parallel.")
```

Comment:
Why day? It's a nice time boundary, but I'm curious if there was reasoning here that makes it the off/on choice, versus being able to specify the range to split on. If someone was using periodic tables with a week range, is there a difference to splitting the query over that time boundary instead?

Reply:
Good question!  The rows in the index are organised by day, and sub day queries pretty much have to read an entire days index entries anyway.  Therefore might as well parallelise by day IMO.

We're running this in prod now and it actually looks like for high-cardinality queries where the execution of the PromQL is the dominant latency, sub-day parallelism might be worth while.  In the future we should probably make this tuneable.